File: android/src/test/java/org/ligi/passandroid/unittest/TheEANHelper.java
package org.ligi.passandroid.unittest;

import java.io.IOException;
import java.net.URISyntaxException;
import org.junit.Test;
import org.ligi.passandroid.ui.edit.EANHelper;
import static org.assertj.core.api.Assertions.assertThat;

public class TheEANHelper {

    @Test
    public void randomEAN13HasCorrectLength() throws URISyntaxException, IOException {
        assertThat(EANHelper.INSTANCE.getRandomEAN13().length()).isEqualTo(13);
    }

    @Test
    public void acceptGoodEAN13() throws URISyntaxException, IOException {
        assertThat(EANHelper.INSTANCE.isValidEAN13("6416016588755")).isTrue();
    }

    @Test
    public void rejectBadEAN13() throws URISyntaxException, IOException {
        assertThat(EANHelper.INSTANCE.isValidEAN13("foo")).isFalse();
    }
}

File: android/src/test/java/org/ligi/passandroid/unittest/TheEANHelper.kt
package org.ligi.passandroid.unittest

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.ligi.passandroid.ui.edit.getRandomEAN13
import org.ligi.passandroid.ui.edit.isValidEAN13

class TheEANHelper {

    @Test
    fun randomEAN13HasCorrectLength() {
        assertThat(getRandomEAN13().length).isEqualTo(13)
    }

    @Test
    fun acceptGoodEAN13() {
        assertThat(isValidEAN13("6416016588755")).isTrue()
    }

    @Test
    fun rejectBadEAN13() {
        assertThat(isValidEAN13("foo")).isFalse()
    }
}
